packages/peregrine/lib/talons/Gallery/useGallery.js

```diff
@@ -0,0 +1,76 @@
+import { useState } from 'react';
+import { useQuery } from '@apollo/client';
+
+import { useUserContext } from '../../context/user';
+import mergeOperations from '../../util/shallowMerge';
+import defaultOperations from './gallery.gql';
+
+export const useGallery = (props = {}) => {
+    const operations = mergeOperations(defaultOperations, props.operations);
+
+    const [{ isSignedIn }] = useUserContext();
+
+    const [currentPage, setCurrentPage] = useState(1);
+
+    const {
+        client,
+        data: { customerWishlistProducts }
+    } = useQuery(operations.getProductsInWishlistsQuery);
+
+    const { data: wishlistConfigData } = useQuery(
+        operations.getWishlistConfigQuery,
+        { fetchPolicy: 'cache-and-network' }
+    );
+
+    useQuery(operations.getWishlistItemsQuery, {
+        fetchPolicy: 'cache-and-network',
+        onCompleted: data => {
+            const itemsToAdd = new Set();
```

Comment:
Might be a dumb question, are you fetching page by page and not all of them at the same time, in the interest of avoiding lag if there are too many items in the list?